playshogi-library-shogi/src/main/java/com/playshogi/library/shogi/models/position/ShogiBoardStateImpl.java

```diff
@@ -24,12 +26,12 @@ private ShogiBoardStateImpl(final int width, final int height, final Piece[][] b
     }
 
     @Override
-    public Piece getPieceAt(final int column, final int row) {
-        return board[column - 1][row - 1];
+    public Optional<Piece> getPieceAt(final int column, final int row) {
```

Comment:
There is some performance cost to returning an optional on such a basic operation: every time that we look at a non-empty square of the board, it will create an object (also resulting in more frequent garbage collection).
Since we are not writing a Shogi engine with this library I think it's fine (I'm not asking to revert the change) - just something to keep in mind

Reply:
Thanks, yes I did consider that there is some slight overhead and some slight code complexity, although protecting against a potential subtle NullPointerException seems worth it, at least for long-lived references or values.